internal/provider/resource/secret_approval_policy.go

```diff
@@ -0,0 +1,419 @@
+}
+
+// secretApprovalPolicyResource is the resource implementation.
+type secretApprovalPolicyResource struct {
+	client *infisical.Client
+}
+
+type SecretApprover struct {
+	Type types.String `tfsdk:"type"`
+	ID   types.String `tfsdk:"id"`
+	Name types.String `tfsdk:"name"`
+}
+
+// secretApprovalPolicyResourceModel describes the data source data model.
+type secretApprovalPolicyResourceModel struct {
+	ID                types.String     `tfsdk:"id"`
+	ProjectID         types.String     `tfsdk:"project_id"`
+	Name              types.String     `tfsdk:"name"`
+	EnvironmentSlug   types.String     `tfsdk:"environment_slug"`
+	SecretPath        types.String     `tfsdk:"secret_path"`
+	Approvers         []SecretApprover `tfsdk:"approvers"`
+	RequiredApprovals types.Int64      `tfsdk:"required_approvals"`
+	EnforcementLevel  types.String     `tfsdk:"enforcement_level"`
+}
+
+// Metadata returns the resource type name.
+func (r *secretApprovalPolicyResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_secret_approval_policy"
+}
+
+// Schema defines the schema for the resource.
+func (r *secretApprovalPolicyResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "Create secret approval policy for your projects",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Description:   "The ID of the secret approval policy",
+				Computed:      true,
+				PlanModifiers: []planmodifier.String{stringplanmodifier.UseStateForUnknown()},
+			},
+			"project_id": schema.StringAttribute{
+				Description: "The ID of the project to add the secret approval policy",
+				Required:    true,
+			},
+			"name": schema.StringAttribute{
+				Description:   "The name of the secret approval policy",
+				Optional:      true,
+				Computed:      true,
+				PlanModifiers: []planmodifier.String{stringplanmodifier.UseStateForUnknown()},
+			},
+			"environment_slug": schema.StringAttribute{
+				Description: "The environment to apply the secret approval policy to",
+				Required:    true,
+			},
+			"secret_path": schema.StringAttribute{
+				Description: "The secret path to apply the secret approval policy to",
+				Required:    true,
+			},
+			"approvers": schema.SetNestedAttribute{
+				Required:    true,
+				Description: "The required approvers",
+				NestedObject: schema.NestedAttributeObject{
+					Attributes: map[string]schema.Attribute{
+						"type": schema.StringAttribute{
+							Description: "The type of approver. Either group or user",
+							Required:    true,
+						},
+						"id": schema.StringAttribute{
+							Description: "The ID of the approver",
+							Optional:    true,
+						},
+						"name": schema.StringAttribute{
+							Description: "The name of the approver. By default, this is the email",
+							Optional:    true,
```

Comment:
Description of the approver should be `The username of the approver. By default, this is the email`. Remember to re-generate docs after making these changes